portal-web/docroot/html/taglib/ui/map/page.jsp

```diff
@@ -19,61 +19,68 @@
 <%
 String protocol = HttpUtil.getProtocol(request);
 
-String apiKey = GetterUtil.getString(request.getAttribute("liferay-ui:map:apiKey"));
 boolean geolocation = GetterUtil.getBoolean(request.getAttribute("liferay-ui:map:geolocation"));
 double latitude = (Double)request.getAttribute("liferay-ui:map:latitude");
 double longitude = (Double)request.getAttribute("liferay-ui:map:longitude");
 String name = GetterUtil.getString((String)request.getAttribute("liferay-ui:map:name"));
 String points = GetterUtil.getString(request.getAttribute("liferay-ui:map:points"));
-String provider = GetterUtil.getString((String)request.getAttribute("liferay-ui:map:provider"));
 
-if (Validator.isNull(provider)) {
-	Group group = layout.getGroup();
+PortletPreferences companyPortletPreferences = PrefsPropsUtil.getPreferences(company.getCompanyId());
 
-	provider = group.getLiveParentTypeSettingsProperty("provider");
+String googleMapsAPIKey = PrefsParamUtil.getString(companyPortletPreferences, request, "googleMapsAPIKey", "");
+String mapsAPIProvider = PrefsParamUtil.getString(companyPortletPreferences, request, "mapsAPIProvider", "Google");
 
-	if (Validator.isNull(provider)) {
-		PortletPreferences companyPortletPreferences = PrefsPropsUtil.getPreferences(company.getCompanyId(), true);
+Group group = themeDisplay.getSiteGroup();
 
-		provider = companyPortletPreferences.getValue("provider", "Google");
-	}
+if (group.isStagingGroup()) {
+	group = group.getLiveGroup();
 }
 
-name = namespace + name;
-%>
+UnicodeProperties groupTypeSettings = new UnicodeProperties();
+
+if (group != null) {
+	groupTypeSettings = group.getTypeSettingsProperties();
+}
 
-<c:if test='<%= provider.equals("Google") %>'>
-	<liferay-util:html-bottom outputKey="js_maps_google_skip_map_loading">
-		<script>
-			Liferay.namespace('Maps').onGMapsReady = function(event) {
-				Liferay.Maps.gmapsReady = true;
+googleMapsAPIKey = PropertiesParamUtil.getString(groupTypeSettings, request, "googleMapsAPIKey", googleMapsAPIKey);
```

Comment:
Same as before. We should read the values directly from groupTypeSettings and not fallback in the request.